homework/index.js

```diff
@@ -16,31 +30,97 @@
+      createAndAppend('td', tr2, { text: `${repoInfo.description}` });
+
+      const tr3 = createAndAppend('tr', tbody);
+      createAndAppend('td', tr3, { text: 'Forks:', class: 'label' });
+      createAndAppend('td', tr3, { text: `${repoInfo.forks}` });
+
+      const tr4 = createAndAppend('tr', tbody);
+      createAndAppend('td', tr4, { text: 'Updated:', class: 'label' });
+      createAndAppend('td', tr4, { text: new Date(repoInfo.updated_at).toLocaleString() });
     });
-    return elem;
+  }
+
+  function renderContributors(repoName, ulContributorInfo) {
+    const url = `https://api.github.com/repos/HackYourFuture/${repoName}/contributors`;
+    fetchJSON(url, (err, contributors) => {
+      ulContributorInfo.innerHTML = '';
+      contributors.forEach(contributor => {
+        const listItems = createAndAppend('li', ulContributorInfo, { class: 'li2' });
+        createAndAppend('img', listItems, {
+          src: contributor.avatar_url,
+          alt: `${contributor.login} photo`,
+        });
+        createAndAppend('h1', listItems);
+        createAndAppend('a', listItems.lastChild, {
+          href: contributor.html_url,
+          target: '_blank',
+          text: `${contributor.login}`,
+        });
+        createAndAppend('p', listItems, { text: `${contributor.contributions}` });
+      });
+    });
+  }
+
+  function createOptionElements(repositories, select) {
+    createAndAppend('option', select, {
+      text: 'Select a repository',
+      disabled: 'disabled',
+    });
+    repositories
+      .sort((a, b) => a.name.localeCompare(b.name))
+      .forEach(repo => {
+        createAndAppend('option', select, { text: repo.name, value: repo.name });
+      });
   }
 
   function main(url) {
-    fetchJSON(url, (err, repositories) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+    const root = document.getElementById('root');
+    const header = createAndAppend('div', root, { class: 'header' });
+    const infoSection = createAndAppend('div', root, { class: 'info-section-div' });
+    const repoInfoSection = createAndAppend('div', infoSection);
+    const contributorInfoSection = createAndAppend('div', infoSection);
+
+    createAndAppend('h1', header, { text: 'HYF Repositories' });
+    const select = createAndAppend('select', header);
+    const ulRepoInfo = createAndAppend('ul', repoInfoSection, { class: 'li-container' });
+    const ulContributorInfo = createAndAppend('ul', contributorInfoSection, {
+      class: 'li-container',
+    });
+
+    fetchJSON(url, (error, data) => {
+      if (error) {
+        createAndAppend('div', root, { text: error.message, class: 'alert-error' });
         return;
       }
-      createAndAppend('pre', root, { text: JSON.stringify(repositories, null, 2) });
+      createOptionElements(data, select);
```

Comment:
Create `option` elements from the sorted array. Take the hint from the assignment to set the value attribute from each option element to the array index of the corresponding repository object, e.g. in your function `createOptionElements`:

```js
repositories.forEach((repo, index) => {
  createAndAppend('option', select, { text: repo.name, value: index });
});
```

Now, in your event hander you can use this index to directly get the repository object that corresponds to the selected repo from the select box like this:

```js
select.addEventListener('change', () => {
  const repo = repositories[select.value];
  renderRepositories(repo, ulRepoInfo);
  renderContributors(repo, ulContributorInfo);
});
```

No need to fetch details for the selected repo again. Just pass the information already in your possession as an argument to the functions that require the information.